rpcserver.go

```diff
@@ -8200,6 +8208,73 @@ func (r *rpcServer) ListAliases(ctx context.Context,
 	return resp, nil
 }
 
+// AddAliases adds a set of aliases to the scid alias map. This is used to
+// create a set of aliases for a base_scid. The response will contain the
+// successfully created mappings.
+func (r *rpcServer) AddAliases(_ context.Context,
+	in *lnrpc.AddAliasesRequest) (*lnrpc.AddAliasesResponse, error) {
+
+	resp := &lnrpc.AddAliasesResponse{
+		AliasMaps: make([]*lnrpc.AliasMap, 0),
+	}
+
+	// Create a map that will store the successfully created scid mappings.
+	scidMap := make(
+		map[lnwire.ShortChannelID]map[lnwire.ShortChannelID]struct{},
+	)
+
+	for _, v := range in.AliasMaps {
+		baseScid := lnwire.NewShortChanIDFromInt(v.BaseScid)
+
+		for _, alias := range v.Aliases {
+			aliasScid := lnwire.NewShortChanIDFromInt(alias)
+
+			err := r.server.aliasMgr.AddLocalAlias(
+				aliasScid, baseScid, false,
+			)
+			if err != nil {
+				rpcsLog.Warnf("Could not create scid alias, "+
+					"base_scid=%v, alias_scid=%v", baseScid,
+					aliasScid)
+
+				continue
+			}
+
+			// If internal map is nil, initialize it.
+			if scidMap[baseScid] == nil {
+				scidMap[baseScid] = make(
+					map[lnwire.ShortChannelID]struct{},
+				)
+			}
+
+			scidMap[baseScid][aliasScid] = struct{}{}
+		}
+	}
+
+	rpcsLog.Info("Created aliases")
+
+	// Now we need to parse the created mappings into an rpc response.
+	for base, aliases := range scidMap {
```

Comment:
nit: could extract this (and the same code from the above `ListAliases` method) into a common `MarshalAliasMap` function in `lnrpc/marshall_utils.go`.